include/mbgl/shaders/gl/shader_group_gl.hpp

```diff
@@ -13,36 +15,54 @@ class ShaderGroupGL final : public gfx::ShaderGroup {
 public:
     ShaderGroupGL(const ProgramParameters& programParameters_)
         : ShaderGroup(),
-          programParameters(programParameters_){};
+          programParameters(programParameters_) {}
     ~ShaderGroupGL() noexcept override = default;
 
     gfx::ShaderPtr getOrCreateShader(gfx::Context& context,
-                                     const std::vector<std::string>& propertiesAsUniforms,
+                                     const std::unordered_set<StringIdentity>& propertiesAsUniforms,
                                      std::string_view firstAttribName) override {
         constexpr auto& name = shaders::ShaderSource<ShaderID, gfx::Backend::Type::OpenGL>::name;
         constexpr auto& vert = shaders::ShaderSource<ShaderID, gfx::Backend::Type::OpenGL>::vertex;
         constexpr auto& frag = shaders::ShaderSource<ShaderID, gfx::Backend::Type::OpenGL>::fragment;
 
-        uint32_t key = 0;
+        // quick-n-dirty order-indepdendent hash combine
+        size_t sum = 0, product = 1;
+        for (const auto nameID : propertiesAsUniforms) {
+            constexpr auto somePrime = 1099511628211ll;
+            sum += nameID;
+            product *= nameID * somePrime;
```

Comment:
+1 extract this to a function. In particular, comment the usage of `somePrime` as the FNV prime (assuming this is where it came from).

Reply:
There's probably a better way to do this and I'd rather replace it entirely, but I don't disagree.  It'll overflow, but that's fine; the results don't need to be ordered, just unique, and come out the same regardless of the input ordering.

This seems a bit lame, especially with the possibility of a collision, but using a `std::set` where each element is allocated separately is also bad.  I was thinking about using a custom class instead of `unordered_set<StringIdentity>` that wraps a vector, since we have ~10-ish elements max, but I'm not sure it's worth the effort.

Reply:
Done.  Thoughts?

Reply:
Looks good.